cdap-api/src/main/java/co/cask/cdap/api/app/Config.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.app;
+
+import co.cask.cdap.api.templates.plugins.PluginConfig;
+
+/**
+ * Configuration class that defines the {@link Application}'s configuration.
+ * This class will replace {@link PluginConfig}. {@link PluginConfig} has been retained for backward compatibility.
+ */
+public class Config extends PluginConfig {
```

Comment:
should be the other way. Move methods from PluginConfig and make PluginConfig extends from this interface.